node/relay_request.js

```diff
@@ -51,11 +52,33 @@ RelayRequest.prototype.createOutRequest = function createOutRequest(host) {
         return;
     }
 
+    self.peer = self.channel.peers.choosePeer(null, {
+        host: host
+    });
+    if (!self.peer) {
+        self.onError(errors.NoPeerAvailable());
+        return;
+    }
```

Comment:
This might be the part that prevents retries. Does erring early when there is no peer available prevent the relay handler from proceeding into retries? Seems like this avoids a very specific case of retrying.